playwright-e2e/lib/component/dsm/navigation/navigation.ts

```diff
@@ -1,33 +1,56 @@
+import {APIRequestContext, Page} from '@playwright/test';
 import Dropdown from 'lib/widget/dsm/dropdown';
 import ParticipantListPage from 'pages/dsm/participantList-page';
-import { MainMenu } from './enums/mainMenu.enum';
-import { StudyNav } from './enums/studyNav.enum';
-import { Study } from './enums/selectStudyNav.enum';
-import { NavigationItems } from './navigation-types';
-import {Miscellaneous} from 'lib/component/dsm/navigation/enums/miscellaneousNav.enum';
+import {MainMenuEnum} from './enums/mainMenu-enum';
+import {StudyNavEnum} from './enums/studyNav-enum';
+import {StudyEnum} from './enums/selectStudyNav-enum';
+import {NavigationItems} from './navigation-types';
+import {SamplesNavEnum} from './enums/samplesNav-enum';
+import KitsWithoutLabelPage from 'pages/dsm/kitsInfo-pages/kitsWithoutLabel-page';
+import InitialScanPage from 'pages/dsm/initialScan-page';
+import FinalScanPage from 'pages/dsm/finalScan-page';
+import KitUploadPage from 'pages/dsm/kitUpload-page/kitUpload-page';
+import {MiscellaneousEnum} from 'lib/component/dsm/navigation/enums/miscellaneousNav-enum';
+import KitsSentPage from 'pages/dsm/kitsInfo-pages/kitsSentPage';
+import KitsReceivedPage from 'pages/dsm/kitsInfo-pages/kitsReceived-page/kitsReceivedPage';
+
+
+type Selection = StudyNavEnum | StudyEnum | SamplesNavEnum | MiscellaneousEnum;
 
 export class Navigation {
   private readonly navigationItems: Partial<NavigationItems> = {
-    study: new Map([[StudyNav.PARTICIPANT_LIST, new ParticipantListPage(this.page)]])
+    study: new Map<string, object>([[StudyNavEnum.PARTICIPANT_LIST, new ParticipantListPage(this.page)]]),
+    samples: new Map<string, object>([
+      [SamplesNavEnum.KITS_WITHOUT_LABELS, new KitsWithoutLabelPage(this.page)],
+      [SamplesNavEnum.INITIAL_SCAN, new InitialScanPage(this.page)],
+      [SamplesNavEnum.FINAL_SCAN, new FinalScanPage(this.page)],
+      [SamplesNavEnum.KIT_UPLOAD, new KitUploadPage(this.page)],
+      [SamplesNavEnum.SENT, new KitsSentPage(this.page)],
+      [SamplesNavEnum.RECEIVED, new KitsReceivedPage(this.page, this.request)],
+    ])
   };
 
-  constructor(private readonly page: Page) {}
+  constructor(private readonly page: Page, private readonly request: APIRequestContext) {}
+
+  public async selectStudy(studyName: StudyEnum): Promise<void> {
+    await this.selectFrom(MainMenuEnum.SELECTED_STUDY, studyName);
+  }
 
-  public async selectStudy(studyName: Study): Promise<void> {
-    await this.selectFrom(MainMenu.SELECTED_STUDY, studyName);
+  public async selectFromSamples<T extends object>(sampleNav: SamplesNavEnum): Promise<T> {
+    await this.selectFrom(MainMenuEnum.SAMPLES, sampleNav);
+    return (this.navigationItems.samples as Map<string, object>).get(sampleNav) as T;
   }
 
-  async selectFromStudy<T extends object>(studyNav: StudyNav): Promise<T> {
-    await this.selectFrom(MainMenu.STUDY, studyNav);
+  public async selectFromStudy<T extends object>(studyNav: StudyNavEnum): Promise<T> {
```

Comment:
Cool generic function. Too bad none of DSM page extends a base dsm common page. something like `<T extends DSMPageBase>`

Reply:
Good idea. We might think. At this moment there is not any field or method that is the same across all DSM pages, except the page itself.